### PR TITLE
feat(MeshTrace): `MeshGateway` support 

### LINKED ISSUE #5231: MeshTrace with MeshGateway
### Description

MeshTrace should work also with MeshGateway

### PR DESCRIPTION
**Pull request by michaelbeaumont**
### Checklist prior to review

- [x] Link to docs PR or issue --
- [x] Link to UI issue or PR --
- [x] Is the [issue worked on linked][1]? -- Closes #5231 
- [x] The PR does not hardcode values that might break projects that depend on kuma (e.g. "kumahq" as a image registry) --
- [x] The PR will work for both Linux and Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Unit Tests -- none
- [x] E2E Tests --
- [x] Manual Universal Tests -- none
- [x] Manual Kubernetes Tests -- none 
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [x] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: test/e2e_env/universal/observability/meshtrace.go [hunk 1]
```diff
@@ -34,12 +37,20 @@ func PluginTest() {
...
+
 	BeforeAll(func() {
 		err := NewClusterSetup().
 			Install(obs.Install(obsDeployment)).
 			Install(MeshUniversal(mesh)).
 			Install(TestServerUniversal("test-server", mesh, WithArgs([]string{"echo", "--instance", "universal1"}))).
 			Install(DemoClientUniversal(AppModeDemoClient, mesh, WithTransparentProxy(true))).
+			Install(GatewayProxyUniversal(mesh, "edge-gateway")).
+			Install(YamlUniversal(gateway.MkGateway("edge-gateway", mesh, false, "example.kuma.io", "test-server", 8080))).
+			Install(gateway.GatewayClientAppUniversal("gateway-client")).
```
Comment (slonka) [inline test/e2e_env/universal/observability/meshtrace.go:53]: "gateway-client" is not listed in the assertion ([code](https://github.com/kumahq/kuma/pull/5243/files#diff-0b974c3ee0b3ce1a922dfa36bdb6d3878f7bf7bee510ea253e7d530f95e4d903R93-R97)), do we need it?

File: test/e2e_env/universal/observability/meshtrace.go [hunk 2]
```diff
@@ -64,10 +75,25 @@ func PluginTest() {
 			}
 			// then traces are published
 			return obsClient.TracedServices()
-		}, "30s", "1s").Should(Equal([]string{
+		}, "30s", "1s").Should(ContainElements([]string{
```
Comment (slonka) [inline test/e2e_env/universal/observability/meshtrace.go:78]: Why do we want to change it? Might there be more `TracedServices` than the 3 mentioned below?

**Review by slonka — COMMENTED** [review]
Looks good, some questions:
1. Should we add a [unit test](https://github.com/kumahq/kuma/blob/1b3ce91d6142df61a1a169860abf419934c6aebf/pkg/plugins/policies/meshtrace/plugin/v1alpha1/plugin_test.go#L25)
2. Why are we not adding a k8s test?

File: test/e2e_env/universal/observability/meshtrace.go (hunk 1 above)
Comment (michaelbeaumont) [reply test/e2e_env/universal/observability/meshtrace.go:53]: it's used by the `ProxySimpleRequests` function.

**Comment by michaelbeaumont** [comment]
> * Should we add a [unit test](https://github.com/kumahq/kuma/blob/1b3ce91d6142df61a1a169860abf419934c6aebf/pkg/plugins/policies/meshtrace/plugin/v1alpha1/plugin_test.go#L25)

I can

> * Why are we not adding a k8s test?

I don't understand the reason we have both for these plugins in the first place actually. Does their behavior depend on whether we're using k8s or universal?

**Comment by slonka** [comment]
> I don't understand the reason we have both for these plugins in the first place actually. Does their behavior depend on whether we're using k8s or universal?

Yeah, was wondering the same. Tracked it down to this PR https://github.com/kumahq/kuma/pull/930/files - maybe @lobkovilya remembers (Jakub is OOO).

**Comment by lahabana** [comment]
Yeah I don't see the value of having e2e tests for both platforms if there's no practical difference in the implem.

**Pull request merged**